File: BE/eos_service/util/response.js
var successObj = (result) => {
	return {
		'status':1,
		'result': result,
	}
}

var failObj = (message, errorCode = 404) => {
	return {
		'status':0,
		'error': {
			'code': errorCode,
			'message': message,
		},
	}
}

var response_express = {
	success: (res, result) => {
		console.log("success", result);
		res.json(successObj(result));
	},

	fail: (res, httpCode, message, errorCode) => {
        console.log("fail", message);
		res.status(httpCode).json(failObj(message, errorCode));
	},
};

module.exports={successObj, failObj, response_express}

File: BE/main_service/util/response.go
package util

import (
	"log"

	"github.com/gin-gonic/gin"
)

type response struct {
}

var Response response

func (response) Success(c *gin.Context, data interface{}) {
	c.JSON(200, data)
}

func (response) Error(c *gin.Context, err MyError) {
	log.Println(err.Raw)
	c.JSON(err.HttpCode, gin.H{
		"code":    err.ErrorCode,
		"message": err.Message,
	})
}

func (response) Abort(c *gin.Context, err MyError) {
	log.Println(err.Raw)
	c.AbortWithStatusJSON(err.HttpCode, gin.H{
		"code":    err.ErrorCode,
		"message": err.Message,
	})
}
